plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/frontend.tsx

```diff
@@ -146,23 +128,27 @@ const { state } = store< Store >( 'woocommerce/product-button', {
 				// After the cart is updated, sync the temporary number of items again.
 				context.temporaryNumberOfItems = state.numberOfItemsInTheCart;
 			} catch ( error ) {
-				const storeNoticeBlock =
-					document.querySelector( storeNoticeClass );
-
-				if ( ! storeNoticeBlock ) {
-					document
-						.querySelector( '.entry-content' )
-						?.prepend( createNoticeContainer() );
-				}
-
-				const domNode =
-					storeNoticeBlock ??
-					document.querySelector( storeNoticeClass );
-
 				const message = ( error as Error ).message;
+				const noticesStore = store< StoreNoticesStore >(
+					'woocommerce/store-notices'
+				);
 
-				if ( domNode ) {
-					injectNotice( domNode, decodeEntities( message ) );
+				// If the user deleted the hooked store notice block, the store won't be present
+				// and we should not add a notice.
+				if ( noticesStore ) {
+					// The old implementation always overwrites the last notice, so
+					// we remove the last notice before adding a new one.
+					if ( state.noticeId !== '' ) {
+						noticesStore.actions.removeNotice( state.noticeId );
+					}
+
+					const noticeId = noticesStore.actions.addNotice( {
+						notice: decodeEntities( message ),
```

Comment:
Is `decodeEntities` really necessary considering that we will end up injecting the message into the HTML using `ref.innerHTML`? It seems like a React leftover.

Take into account that we also need to get rid of `@wordpress/html-entities`.

Reply:
Oh yes you're absolutely right, I should have spotted that. Actually the console error should probably decode entities to make it more readable but I don't think I'll do that here, as you say we also need to get rid of that dependency.

Resolved in https://github.com/woocommerce/woocommerce/pull/54552/commits/a867a76c3fcb0d987b386923f6833fe9b7ee01b1